addons/mail/models/mail_thread.py

```diff
@@ -1091,10 +1195,14 @@ def message_process(self, model, message, custom_values=None,
                          msg_dict.get('email_from'), msg_dict.get('to'), msg_dict.get('message_id'))
             return False
 
+        if self._routing_detect_loop_from_headers(message):
```

Comment:
By the way why not giving msg_dict with parsed values ?

Reply:
Indeed we have to use both (message for new headers, msg_dict for already parsed values). Could you just keep the same ordering of parameters than in other route methods ? Sometimes message is first, sometimes msg_dict :) . Seems we always have message, msg_dict currently (aka :socks: ).

Reply:
@tde-banana-odoo Done :)